mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -32,42 +31,40 @@ class Error(Exception):
 
 
 class SnippetClient(jsonrpc_client_base.JsonRpcClientBase):
-    def __init__(self, package, port, adb_proxy):
+    def __init__(self, package, host_port, adb_proxy):
         """Initialzies a SnippetClient.
   
         Args:
           package: (str) The package name of the apk where the snippets are
             defined.
-          port: (int) The port at which to start the snippet client. Note that
-            the same port will currently be used for both the device and host
-            side of the connection.
+          host_port: (int) The port at which to start the snippet client. Note
+            that the same port will currently be used for both the device and
+            host side of the connection.
             TODO(adorokhine): allocate a distinct free port for both sides of
             the connection; it is not safe in general to reuse the same one.
           adb_proxy: (adb.AdbProxy) The adb proxy to use to start the app.
         """
-        super(SnippetClient, self).__init__(adb_proxy)
+        super(SnippetClient, self).__init__(host_port, adb_proxy)
         self.app_name = package
-        self._package = package
-        self._port = port
-
-    @property
-    def package(self):
-        return self._package
+        # TODO(adorokhine): Don't assume that a free host-side port is free on
+        # the device as well. Both sides should allocate a unique port.
+        self.device_port = host_port
+        self.package = package
```

Comment:
There's no point to trying to do access control in python because you can 'accidentally' change literally anything. You can override another object's functions, imports, modules, even the setattr function itself. You can change self._package which underlies this property.

What makes self.package more important to protect than, say, self.stop_app?

I just can't get behind trying to access control one or two random attributes in python at the expense of more code, when there's hundreds of other ways anyone could accidentally or purposefully break our object, especially when it's a property the user controls anyway (since they pass it into the constructor).

It doesn't buy us anything but more verbose code; there's just no point to it.